diff --git a/pkgs/applications/graphics/djview/default.nix b/pkgs/by-name/dj/djview/package.nix
similarity index 85%
rename from pkgs/applications/graphics/djview/default.nix
rename to pkgs/by-name/dj/djview/package.nix
index bf8ccce9fe00..557c70912e13 100644
--- a/pkgs/applications/graphics/djview/default.nix
+++ b/pkgs/by-name/dj/djview/package.nix
@@ -52,16 +52,27 @@ stdenv.mkDerivation rec {
     "--with-x"
     "--with-tiff"
     "--disable-nsdejavu" # 2023-11-14: modern browsers have dropped support for NPAPI
-  ] ++ lib.optional stdenv.hostPlatform.isDarwin "--enable-mac";
+  ];
+
+  postInstall =
+    let
+      Applications = "$out/Applications";
+    in
+    lib.optionalString stdenv.hostPlatform.isDarwin ''
+      mkdir -p ${Applications}
+      cp -a src/djview.app -t ${Applications}
+    '';
 
   meta = with lib; {
-    broken = stdenv.hostPlatform.isDarwin;
-    description = "Portable DjVu viewer (Qt5) and browser (nsdejavu) plugin";
+    description = "Portable DjVu viewer (Qt5)";
     mainProgram = "djview";
     homepage = "https://djvu.sourceforge.net/djview4.html";
     license = licenses.gpl2Plus;
     platforms = platforms.unix;
-    maintainers = with maintainers; [ Anton-Latukha ];
+    maintainers = with maintainers; [
+      Anton-Latukha
+      bryango
+    ];
     longDescription = ''
       The portable DjVu viewer (Qt5) and browser (nsdejavu) plugin.
 
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index d775e24bc64a..4788211ab2f2 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -12149,7 +12149,6 @@ with pkgs;
 
   djv = callPackage ../by-name/dj/djv/package.nix { openexr = openexr_2; };
 
-  djview = callPackage ../applications/graphics/djview { };
   djview4 = djview;
 
   dmenu = callPackage ../applications/misc/dmenu { };
