diff --git a/pkgs/by-name/ju/jujutsu/package.nix b/pkgs/by-name/ju/jujutsu/package.nix
index 5216fd971c1a..be68e89d709c 100644
--- a/pkgs/by-name/ju/jujutsu/package.nix
+++ b/pkgs/by-name/ju/jujutsu/package.nix
@@ -14,16 +14,16 @@
 
 rustPlatform.buildRustPackage (finalAttrs: {
   pname = "jujutsu";
-  version = "0.36.0";
+  version = "0.36.0-unstable-2025-12-22";
 
   src = fetchFromGitHub {
     owner = "jj-vcs";
     repo = "jj";
-    tag = "v${finalAttrs.version}";
-    hash = "sha256-HGMzNXm6vWKf/RHPwB/soDqxAvCOW1J6BPs0tsrEuTI=";
+    rev = "1e521f564dcec8f93ba3b0a494259d38ef0d8341";
+    hash = "sha256-/AsSdKo/iZ8Ub/5ljB6FL+/3o/0lsa/pIodEcqbUfzs=";
   };
 
-  cargoHash = "sha256-jai0FNuCUcgN+ZmmYgbFrMK1Z1vcv21wALkEb74h7H0=";
+  cargoHash = "sha256-alfaTAgtJLRGI34Bz3ylhY84GHJTqJ0Qg+OqLkU4OR4=";
 
   nativeBuildInputs = [
     installShellFiles
